lib/puppet/provider/sensuctl.rb

```diff
@@ -117,6 +128,17 @@ def self.sensuctl_auth_types()
     auth_types
   end
 
+  def self.dump(resource_type)
+    output = sensuctl(['dump',resource_type,'--format','yaml','--all-namespaces'])
+    Puppet.debug("YAML dump of #{resource_type}:\n#{output}")
+    resources = []
+    dumps = output.split('---')
+    dumps.each do |d|
+      resources << YAML.load(d)
```

Comment:
why are we doing yaml output here though using json output above?

Reply:
Because `sensuctl dump` doesn't support `--format json`. It supports `wrapped-json` and `yaml` and neither can be parsed by Ruby without hacks and easier to split the YAML output on `---` and then ingest each item.

Reply:
Isn't there an open issue you raised against sensu-go related to this? Suggest commenting that this is done because of what you noted and if there is an issue link add that, too.

Reply:
Sort of? The issue I opened pre-dated the `dump` command. Taking the YAML and splitting on `---` isn't that difficult or expensive. This feature works regardless. I'd rather not wait on upstream to support something we can work around. Opened new issue for this but I think this PR should be good to go without upstream making our lives easier.

https://github.com/sensu/sensu-go/issues/3424

Reply:
Also #1175 uses the dump + YAML which is all we can do for that PR. Just the behavior of upstream and we can remove the YAML code if / when upstream updates to support json for dump.

Reply:
Using the split and yaml is fine, I just want to document why we do it next to the code along with the link to the issue you noted.

Reply:
Added comments to all locations where we execute `sensuctl dump`.